### ISSUE TITLE
'Ignored' 0 values when creating RFC string.

### ISSUE THREAD

**Issue by katy-deeson** [rlanvin/php-rrule#49]
My team noticed an issue when using this library to create RRules, in that when you create a rule from an array of values that contain parts with a value of `0`, and then use `RRule->rfcString()`, the items with a value of `0` are dropped.

For example:

```
$rrule = new RRule(["FREQ" => "DAILY", "BYMINUTE" => 0, "BYHOUR" => 0]);
echo $rrule->rfcString();
```

Actual: `FREQ=DAILY;`
Expected value: `FREQ=DAILY;BYMINUTE=0;BYHOUR=0;`

This doesn't make any difference when figuring out recurrences, etc. as 0 would be a default value. However, the lack of a value is playing havoc within our application as we treat `null` values differently from `0` values.

I think this line of code is the culprit:

https://github.com/rlanvin/php-rrule/blob/master/src/RRule.php#L640

Replacing `if ( $value )` with `if ( isset($value) )` would resolve the issue. Happy to submit a PR. :)

**Comment by rlanvin** [comment]
Good catch. It *does* make a difference - a `BYHOUR` of 0 returns occurrences at 0 hour, whereas no `BYHOUR` (null) takes the hour part of `DTSTART`. For HOURLY, MINUTELY and SECONDLY this actually completely change the results. So yeah, it's not only your application, it's definitely a problem.

> Happy to submit a PR. :)

Please do! However I would use `if ( $value !== null )` instead of `isset` since it's a bit clearer (we know the `$value` is set, we just want to check if it's null). It would be perfect if you could also write a unit test for this bug in the same PR.
